internal/policy/container/has_modified_files_test.go

```diff
@@ -0,0 +1,214 @@
+		})
+		When("there is a modified RPM file found", func() {
+			var pkgs map[string]packageFilesRef
+			BeforeEach(func() {
+				pkgs = pkgRef
+				pkgSecondLayer := pkgRef["secondlayer"]
+				pkgSecondLayer.LayerFiles = append(pkgs["secondlayer"].LayerFiles, "this")
+				pkgs["secondlayer"] = pkgSecondLayer
+			})
+			It("should not pass Validate", func() {
+				ok, err := hasModifiedFiles.validate(context.Background(), layers, pkgs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeFalse())
+			})
+		})
+		When("a package is updated", func() {
+			var pkgs map[string]packageFilesRef
+			BeforeEach(func() {
+				pkgs = pkgRef
+				pkgSecondLayer := pkgs["secondlayer"]
+				pkgSecondLayerPackageFiles := pkgSecondLayer.LayerPackageFiles
+				pkgSecondLayerPackageFiles["this"] = "foo-2.0"
+				pkgSecondLayer.LayerPackageFiles = pkgSecondLayerPackageFiles
+				pkgs["secondlayer"] = pkgSecondLayer
+			})
+			It("should pass validate", func() {
+				ok, err := hasModifiedFiles.validate(context.Background(), layers, pkgs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeTrue())
+			})
+		})
+		When("a package is removed", func() {
+			var pkgs map[string]packageFilesRef
+			BeforeEach(func() {
+				pkgs = pkgRef
+				pkgSecondLayer := pkgs["secondlayer"]
+				pkgSecondLayerPackageFiles := pkgs["secondlayer"].LayerPackageFiles
+				delete(pkgSecondLayerPackageFiles, "this")
+				pkgSecondLayer.LayerPackageFiles = pkgSecondLayerPackageFiles
+				pkgs["secondlayer"] = pkgSecondLayer
+			})
+			It("should pass validate", func() {
+				ok, err := hasModifiedFiles.validate(context.Background(), layers, pkgs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeTrue())
+			})
+		})
+	})
+
+	When("the first layer is empty", func() {
+		var zeroPkgRef map[string]packageFilesRef
+		var zeroLayers []string
+		BeforeEach(func() {
+			zeroPkgRef = pkgRef
+			zeroPkgRef["zerolayer"] = packageFilesRef{
+				LayerFiles:        []string{},
+				LayerPackages:     make(map[string]struct{}),
+				LayerPackageFiles: make(map[string]string),
+				HasRPMDB:          false,
+			}
+			zeroLayers = append([]string{"zerolayer"}, layers...)
+		})
+		It("should ignore it", func() {
+			ok, err := hasModifiedFiles.validate(context.Background(), zeroLayers, zeroPkgRef)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ok).To(BeTrue())
+		})
+	})
+
+	When("building the installed file list for installed packages", func() {
+		const (
+			basename = "foobasename"
+			dirname  = "foodirname"
+			dirindex = 0
+		)
+		var goodPkgList []*rpmdb.PackageInfo
+		var badPkgList []*rpmdb.PackageInfo
+
+		BeforeEach(func() {
+			goodPkgList = []*rpmdb.PackageInfo{
+				{
+					BaseNames:  []string{basename},
+					DirIndexes: []int32{dirindex},
+					DirNames:   []string{dirname},
+				},
+			}
+
+			badPkgList = []*rpmdb.PackageInfo{
+				{
+					BaseNames:  []string{basename},
+					DirIndexes: []int32{dirindex},
+					DirNames:   []string{dirname, "extra"},
+				},
```

Comment:
I get that the intent is to not have a `BeforeEach` in each test, but I'm not sure I love this style where we setup all the data, then only use one half in one test and other half in another. Maybe it's okay in this use case since we are testing individual methods, and not an entire flow.

Reply:
I see what you mean. Yeah, I can fix that.